test/core/event_engine/cf/cf_engine_test.cc

```diff
@@ -80,6 +82,211 @@ TEST(CFEventEngineTest, TestConnectionCancelled) {
   client_signal.WaitForNotification();
 }
 
+namespace {
+std::vector<std::string> ResolvedAddressesToStrings(
+    const std::vector<EventEngine::ResolvedAddress> addresses) {
+  std::vector<std::string> ip_strings;
+  std::transform(addresses.cbegin(), addresses.cend(),
+                 std::back_inserter(ip_strings), [](auto const& address) {
+                   return ResolvedAddressToString(address).value_or("ERROR");
+                 });
+  return ip_strings;
+}
+}  // namespace
+
+TEST(CFEventEngineTest, CFEventEngineTest_TestCreateDNSResolver_Test) {
+  grpc_core::MemoryQuota memory_quota("cf_engine_test");
+  auto cf_engine = std::make_shared<CFEventEngine>();
+
+  EXPECT_TRUE(cf_engine->GetDNSResolver({}).status().ok());
+  EXPECT_TRUE(cf_engine->GetDNSResolver({.dns_server = ""}).status().ok());
+  EXPECT_EQ(
+      cf_engine->GetDNSResolver({.dns_server = "8.8.8.8"}).status().code(),
+      absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(
+      cf_engine->GetDNSResolver({.dns_server = "8.8.8.8:53"}).status().code(),
+      absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(
+      cf_engine->GetDNSResolver({.dns_server = "invalid"}).status().code(),
+      absl::StatusCode::kInvalidArgument);
+}
+
+TEST(CFEventEngineTest, TestResolveLocalhost) {
+  grpc_core::Notification resolve_signal;
+
+  auto cf_engine = std::make_shared<CFEventEngine>();
+  auto dns_resolver = cf_engine->GetDNSResolver({});
+
+  dns_resolver.value()->LookupHostname(
+      [&resolve_signal](auto result) {
+        EXPECT_TRUE(result.status().ok());
+        EXPECT_THAT(ResolvedAddressesToStrings(result.value()),
+                    testing::UnorderedElementsAre("127.0.0.1:80", "[::1]:80"));
```

Comment:
Mostly out of curiosity, I'm not an iOS dev: are all iOS test environments guaranteed to support ipv4 and ipv6?

Reply:
I don't know. I don't have a ipv4 or ipv6 only network environment and somehow have the simulator or an iOS device to connect to.